app/uk/gov/hmrc/merchandiseinbaggage/service/CalculationService.scala

```diff
@@ -33,6 +35,21 @@ class CalculationService @Inject()(mibConnector: MibConnector)(implicit ec: Exec
   def paymentCalculations(importGoods: Seq[ImportGoods])(implicit hc: HeaderCarrier): Future[CalculationResults] =
     mibConnector.calculatePayments(importGoods.map(_.calculationRequest)).map(withLogging)
 
+  def thresholdCheck(declarationJourney: DeclarationJourney)(implicit hc: HeaderCarrier): OptionT[Future, Boolean] =
+    if (declarationJourney.amendmentIfRequiredAndComplete.isDefined)
+      isAmendPlusOriginalOverThreshold(declarationJourney)
+    else OptionT.pure[Future](false)
+
+  private def isAmendPlusOriginalOverThreshold(declarationJourney: DeclarationJourney)(
+    implicit hc: HeaderCarrier): OptionT[Future, Boolean] =
+    for {
+      amendments                 <- OptionT.fromOption[Future](declarationJourney.amendmentIfRequiredAndComplete)
+      calculationResults         <- OptionT.liftF(paymentCalculations(amendments.goods.importGoods))
+      originalDeclaration        <- OptionT(mibConnector.findDeclaration(declarationJourney.declarationId))
+      originalCalculationResults <- OptionT.fromOption[Future](originalDeclaration.maybeTotalCalculationResult)
+      newThreshold = calculationResults.totalGbpValue.value + originalCalculationResults.totalGbpValue.value
```

Comment:
probably should be called something like `totalGbpAmount` or something, the threshold is a static amount, this is the value we are comparing against the threshold.